src/common/prx-navigation.js

```diff
@@ -12,6 +12,14 @@ angular.module('prxNavigation', ['ui.router'])
   }
 
   function instrument(route, scope) {
+    if (route.data && route.data.loginRequired &&
+      !route.data.$loginInstrumented) {
+      route.resolve = route.resolve || {};
+      route.data.$loginInstrumented = true;
+      route.resolve.currentUser = ['PrxAuth', function (PrxAuth) {
+        return PrxAuth.currentUser(true);
+      }];
+    }
```

Comment:
very simple, if the state's data hash has `'loginRequired': true`, we automatically add a rejecting-on-failure promise for `currentUser` to that state.

In the future, I'll add something which watches for failed state transitions based on this promise and instead of reverting to the previous state, throw up a login prompt and retry the state transition once logged in successfully.

This should probably be moved to the auth module, but again sort of crunching. It'll be easy to move it there at the time that I add the state transition listener mentioned above.